metis/lib/server/controllers/upload_controller.rb

```diff
@@ -71,7 +80,19 @@ def upload_start
       metis_uid: metis_uid,
     ).first
 
-    raise Etna::BadRequest, 'No matching upload' unless upload
+    if !upload
+      require_params(:project_name, :bucket_name, :file_path)
+      hmac = @request.env['etna.hmac']
+      upload = create_upload({
+        file_path: @params[:file_path],
```

Comment:
Hmm, I am confused by this - above in authorize we create the upload; here we find an existing upload, then if it does not exist we call create_upload - if we can do that safely, doesn't this mean we no longer need to do it in authorize at all?

Also - here you call create_upload, which uses `find_or_create`, but we know there is no upload to be found, so create() suffices - but can we not just dispense with the Upload.where above and use create_upload just once in this controller?

Reply:
Oh yeah, I see what you're saying. We could move Upload creation to `upload_start` in all cases, since it's not really required in `authorize` -- there we could just do the checks and return the URL.

Reply:
@graft , like so?